src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -0,0 +1,599 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2020 Andrew Mickelson
+//               2013 Jonathan De Wachter (dewachter.jonathan@gmail.com)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/DRM/DRMContext.hpp>
+#include <SFML/Window/DRM/WindowImplDRM.hpp>
+#include <SFML/OpenGL.hpp>
+#include <SFML/System/Err.hpp>
+#include <SFML/System/Sleep.hpp>
+#include <drm-common.h>
+#include <cstdlib>
+#include <cstring>
+#include <errno.h>
+#include <poll.h>
+#include <unistd.h>
+
+namespace
+{
+    bool initialized = false;
+    struct drm my_drm;
+    drmEventContext my_evctx;
+    pollfd my_fds;
+    struct gbm_device* my_gbm_device = NULL;
+    int context_count = 0;
+    EGLDisplay display = EGL_NO_DISPLAY;
+    int waiting_for_flip = 0;
+
+    static void page_flip_handler(int fd, unsigned int frame,
+        unsigned int sec, unsigned int usec, void* data)
+    {
+        // suppress unused param warning
+        (void)fd, (void)frame, (void)sec, (void)usec;
+
+        int* temp = static_cast<int*>(data);
+        *temp = 0;
+    }
+
+    static bool wait_for_flip(int timeout)
+    {
+        while (waiting_for_flip)
+        {
+            my_fds.revents = 0;
+
+            if (poll(&my_fds, 1, timeout) < 0)
+                return false;
+
+            if (my_fds.revents & (POLLHUP | POLLERR))
+                return false;
+
+            if (my_fds.revents & POLLIN)
+            {
+                drmHandleEvent(my_drm.fd, &my_evctx);
+            }
+            else
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    void cleanup()
```

Comment:
cleanUp

Reply:
I think `cleanup` makes sense since "cleanup" is one word in this context.